File: packages/sdks/client-sdks/gatrix-js-client-sdk/src/events.ts
/**
 * Event constants for Gatrix Client SDK
 * All events use the 'flags.' prefix for namespacing
 */
export const EVENTS = {
  /** SDK initialized (from storage/bootstrap) */
  FLAGS_INIT: 'flags.init',
  /** First successful fetch completed */
  FLAGS_READY: 'flags.ready',
  /** Started fetching flags from server */
  FLAGS_FETCH: 'flags.fetch',
  /** Started fetching flags from server (alias for FETCH) */
  FLAGS_FETCH_START: 'flags.fetch_start',
  /** Successfully fetched flags from server */
  FLAGS_FETCH_SUCCESS: 'flags.fetch_success',
  /** Error occurred during fetching */
  FLAGS_FETCH_ERROR: 'flags.fetch_error',
  /** Completed fetching flags (success or error) */
  FLAGS_FETCH_END: 'flags.fetch_end',
  /** Flags changed from server */
  FLAGS_CHANGE: 'flags.change',
  /** General SDK error occurred */
  SDK_ERROR: 'flags.error',
  /** Flag accessed (if impressionData enabled) */
  FLAGS_IMPRESSION: 'flags.impression',
  /** Flags synchronized (explicitSyncMode) */
  FLAGS_SYNC: 'flags.sync',
  /** SDK recovered from error state */
  FLAGS_RECOVERED: 'flags.recovered',
  /** Metrics sent to server */
  FLAGS_METRICS_SENT: 'flags.metrics_sent',
  /** Error sending metrics */
  FLAGS_METRICS_ERROR: 'flags.metrics_error',
} as const;

export type EventType = (typeof EVENTS)[keyof typeof EVENTS];

File: packages/sdks/client-sdks/gatrix-unity-sdk/Runtime/Events.cs
// Event constants for Gatrix Unity Client SDK
// All events use the 'flags.' prefix for namespacing

namespace Gatrix.Unity.SDK
{
    public static class GatrixEvents
    {
        /// <summary>SDK initialized (from storage/bootstrap)</summary>
        public const string FlagsInit = "flags.init";

        /// <summary>First successful fetch completed</summary>
        public const string FlagsReady = "flags.ready";

        /// <summary>Started fetching flags from server</summary>
        public const string FlagsFetch = "flags.fetch";

        /// <summary>Started fetching flags from server (alias for Fetch)</summary>
        public const string FlagsFetchStart = "flags.fetch_start";

        /// <summary>Successfully fetched flags from server</summary>
        public const string FlagsFetchSuccess = "flags.fetch_success";

        /// <summary>Error occurred during fetching</summary>
        public const string FlagsFetchError = "flags.fetch_error";

        /// <summary>Completed fetching flags (success or error)</summary>
        public const string FlagsFetchEnd = "flags.fetch_end";

        /// <summary>Flags changed from server</summary>
        public const string FlagsChange = "flags.change";

        /// <summary>General SDK error related to flags</summary>
        public const string FlagsError = "flags.error";

        /// <summary>Flag accessed (if impressionData enabled)</summary>
        public const string FlagsImpression = "flags.impression";

        /// <summary>Flags synchronized (explicitSyncMode)</summary>
        public const string FlagsSync = "flags.sync";

        /// <summary>One or more flags removed from server. Payload: string[] of removed flag names</summary>
        public const string FlagsRemoved = "flags.removed";

        /// <summary>SDK recovered from error state</summary>
        public const string FlagsRecovered = "flags.recovered";

        /// <summary>Metrics sent to server</summary>
        public const string FlagsMetricsSent = "flags.metrics_sent";

        /// <summary>Error sending metrics</summary>
        public const string FlagsMetricsError = "flags.metrics_error";

        /// <summary>Get the per-flag change event name</summary>
        public static string FlagChange(string flagName) => $"flags.{flagName}.change";
    }

    /// <summary>Delegate for Gatrix SDK events</summary>
    public delegate void GatrixEventHandler(object[] args);

    /// <summary>Delegate for universal Gatrix SDK event listeners (OnAny)</summary>
    public delegate void GatrixAnyEventHandler(string eventName, object[] args);

    /// <summary>Delegate for feature flag change watchers</summary>
    public delegate void GatrixFlagWatchHandler(FlagProxy flag);
}
